src/library/dao/directorydao.h

```diff
@@ -28,8 +29,14 @@ class DirectoryDAO : public DAO {
     };
     RemoveResult removeDirectory(const mixxx::FileInfo& oldDir) const;
 
+    enum class RelocateResult {
+        Ok,
+        InvalidOrMissingDirectory,
+        UnreadableDirectory,
+        SqlError,
+    };
     // TODO: Move this function out of the DAO
-    QList<RelocatedTrack> relocateDirectory(
+    std::pair<RelocateResult, QList<RelocatedTrack>> relocateDirectory(
```

Comment:
This could also be a struct to make it more compact, but that appearantly that requires some extras to make it work with `std::tie` in TrackCollection.
Didn't test it.
```c++
struct RelocateResult {
    Result result;
    QList<RelocatedTrack> trackList;
    template<typename T, typename U>
    operator std::tuple<T, U>();
};
```